bottlecap/src/lifecycle/invocation/triggers/sqs_event.rs

```diff
@@ -198,6 +202,81 @@ impl Trigger for SqsRecord {
     }
 }
 
+// extractTraceContextfromAWSTraceHeader extracts trace context from the
+// AWSTraceHeader directly. Unlike the other carriers in this file, it should
+// not be passed to the tracer.Propagator, instead extracting context directly.
+pub(crate) fn extract_trace_context_from_aws_trace_header(
+    value: Option<String>,
+) -> Result<SpanContext, String> {
+    if value.is_none() {
+        return Err("awstrace_header is empty".to_string());
+    }
+    let value = value.unwrap();
+    if !value.starts_with("Root=") {
+        return Err("awstrace_header mismatch".to_string());
+    }
+
+    let mut start_part = 0;
+    let mut trace_id = String::new();
+    let mut parent_id = String::new();
+    let mut sampled = String::new();
+
+    let length = value.len();
+    while start_part < length {
+        let end_part = value[start_part..]
+            .find(';')
+            .map_or(length, |i| i + start_part);
+        let part = &value[start_part..end_part];
+
+        if part.starts_with("Root=") {
+            if trace_id.is_empty() {
+                trace_id = part[24..].to_string();
+            }
+        } else if part.starts_with("Parent=") {
+            if parent_id.is_empty() {
+                parent_id = part[7..].to_string();
+            }
+        } else if part.starts_with("Sampled=") && sampled.is_empty() {
+            sampled = part[8..].to_string();
+        }
+
+        if !trace_id.is_empty() && !parent_id.is_empty() && !sampled.is_empty() {
+            break;
+        }
+        start_part = end_part + 1;
+    }
+
+    let trace_id = u64::from_str_radix(&trace_id, 16)
+        .map_err(|e| format!("Failed to parse trace ID from awstrace_header: {e}"))?;
+    let parent_id = u64::from_str_radix(&parent_id, 16)
+        .map_err(|e| format!("Failed to parse parent ID from awstrace_header: {e}"))?;
+
+    if trace_id == 0 || parent_id == 0 {
+        return Err("awstrace_header contains empty trace or parent ID".to_string());
+    }
+
+    let sampling_priority = i8::from(sampled == "1");
+
+    let mut trace_context = HashMap::new();
+    trace_context.insert("trace_id".to_string(), trace_id.to_string());
+    trace_context.insert("parent_id".to_string(), parent_id.to_string());
+    trace_context.insert(
+        "sampling_priority".to_string(),
+        sampling_priority.to_string(),
+    );
+    Ok(SpanContext {
+        trace_id,
+        span_id: parent_id,
+        sampling: Some(Sampling {
+            priority: Some(sampling_priority),
+            mechanism: None,
+        }),
+        origin: None,
+        tags: HashMap::new(),
```

Comment:
How are you handling 128 bits here?

Reply:
I didn't see any code handling 128 bits for the aws header. I think the datadog-specific format doesn't require 128 bits.
@purple4reina do you have context on this?

Reply:
The upstream tracer may include the upper 64 bits of a trace ID. e.g. https://github.com/DataDog/dd-trace-java/blob/master/dd-java-agent/instrumentation/aws-java-sqs-1.0/src/main/java/datadog/trace/instrumentation/aws/v1/sqs/SqsInterceptor.java#L44

We have a backwards compatibility layer that allows flame graphs to connect 128 bit trace chunks to 64 bit trace chunks given that the lower 64 bits match, but the right thing to do (whether now or in a fast-follow item) is to support 128 bit trace IDs.

Reply:
Thanks!